compiler/rustc_lint/src/late.rs

```diff
@@ -417,33 +405,11 @@ fn late_lint_crate<'tcx, T: LateLintPass<'tcx>>(tcx: TyCtxt<'tcx>, builtin_lints
     let mut passes =
         unerased_lint_store(tcx).late_passes.iter().map(|p| (p)(tcx)).collect::<Vec<_>>();
 
-    if !tcx.sess.opts.unstable_opts.no_interleave_lints {
-        if !passes.is_empty() {
-            late_lint_pass_crate(tcx, LateLintPassObjects { lints: &mut passes[..] });
-        }
-
-        late_lint_pass_crate(tcx, builtin_lints);
-    } else {
-        for pass in &mut passes {
-            tcx.sess.prof.verbose_generic_activity_with_arg("run_late_lint", pass.name()).run(
-                || {
-                    late_lint_pass_crate(tcx, LateLintPassObjects { lints: slice::from_mut(pass) });
-                },
-            );
-        }
-
-        let mut passes: Vec<_> =
-            unerased_lint_store(tcx).late_module_passes.iter().map(|pass| (pass)(tcx)).collect();
-
-        for pass in &mut passes {
-            tcx.sess
-                .prof
-                .verbose_generic_activity_with_arg("run_late_module_lint", pass.name())
-                .run(|| {
-                    late_lint_pass_crate(tcx, LateLintPassObjects { lints: slice::from_mut(pass) });
-                });
-        }
+    if !passes.is_empty() {
+        late_lint_pass_crate(tcx, LateLintPassObjects { lints: &mut passes[..] });
     }
+
+    late_lint_pass_crate(tcx, builtin_lints);
 }
```

Comment:
Can this function be merged with `late_lint_mod`?

Reply:
I'm not sure which function is "this function", because it looks to me like this comment is on `late_lint_mod`. But as mentioned elsewhere, based on your other suggestions I was able to merge `late_lint_mod_pass` into `late_lint_mod`, and also `late_lint_pass_crate` into `late_lint_pass`.

It's not possible to merge `late_lint_mod` and `late_lint_crate`, though.